cli/kubectl_kadalu/logs.py

```diff
@@ -1,40 +1,82 @@
 """
 'logs' subcommand for kubectl-kadalu CLI tool
 """
-#To prevent Py2 to interpreting print(val) as a tuple.
+# To prevent Py2 to interpreting print(val) as a tuple.
 from __future__ import print_function
 
+import sys
+import argparse
 import utils
 
+
 # TODO: provide 'hint' based on grepping error logs. That way, users get to
 #       check for few common errors easily
 
 def set_args(name, subparsers):
     """ add arguments to argparser """
-    # TODO: in future allow users to pass optional pod names
-    # TODO: make 'all-containers' optional based on container name (along with pod name).
     # TODO: provide options to pass options to kubectl logs (like tail, container name etc)
-    parser = subparsers.add_parser(name)
+
+    parser = subparsers.add_parser(
+        name,
+        formatter_class=argparse.RawTextHelpFormatter
+    )
+
+    arg = parser.add_argument
+
+    arg("-p", "--podname",
+        help="Specify pod name to get log info")
+
+    arg("-c", "--container",
+        help="Specify container name to get log info.\n"
+             "To be used along with '--podname'")
+
+    arg("-A", "--allcontainers", action="store_true",
+        help=("Show logs of all containers"
+              "of a particular pod.\n"
+              "To be used along with '--podname'"))
+
     utils.add_global_flags(parser)
 
 
-def validate(_args):
-    """No validation available"""
-    return
+def validate(args):
+    """Validate optional arguments"""
+
+    if args.container and not args.podname:
+        print("Specify pod name for the container '%s'." %args.container)
+        sys.exit(1)
+
+    if args.allcontainers and not args.podname:
+        print("Specify pod name for displaying of all containers.\n"
+              "Use without any optional arguments to get logs of "
+              "all pods and all of it's containers.")
+        sys.exit(1)
+
+    if args.allcontainers and args.container:
+        print("Cannot use both --allcontainers & --container together. "
+              "Choose any one of the above options.")
+        sys.exit(1)
 
 
 def run(args):
     """ perform log subcommand """
 
     try:
-        cmd = utils.kubectl_cmd(args) + ["get", "pods", "-nkadalu", "-oname"]
-        resp = utils.execute(cmd)
-        pods = resp.stdout.split('\n')
+
+        pods, container = [args.podname], "--all-containers=true"
+
+        if args.container:
+            container = '-c' + args.container
+
+        if not args.podname:
+            cmd = utils.kubectl_cmd(args) + ["get", "pods", "-nkadalu", "-oname"]
+            resp = utils.execute(cmd)
+            # Remove empty lines(pod-names) from command response
+            pods = [pod for pod in resp.stdout.split('\n') if pod.strip()]
```

Comment:
Looks neat! Almost there, can you give a try with `resp.stdout.strip('\n').split('\n')` and remove conditional in list comprehension?

Reply:
Tried but it does not remove the empty spaces

```python3
pods = "pod1\npod2\n\npod3\npod4\n\n\npod5"

print(pods.strip('\n'))

pods = [pod for pod in pods.strip('\n').split('\n')]

print(pods)
```
O/P
```bash
vatsa@vatsa-VirtualBox:~/tryKadalu/test$ python3 test.py
pod1
pod2

pod3
pod4


pod5
['pod1', 'pod2', '', 'pod3', 'pod4', '', '', 'pod5']
```

Reply:
Typo from my side, you can use `strip` and `split` without any args to remove any type of multuple spaces (\t \s \n etc). So you just use `resp.stdout.split()`

Reply:
Done!